mathics/core/definitions.py

```diff
@@ -243,6 +244,9 @@ def get_current_context(self):
     def get_context_path(self):
         return self.context_path
 
+    def get_inputfile(self):
+        return self.inputfile
```

Comment:
```
         return self.inputfile if hasattr(self, "inputfile") else ""
```

This works even if the attribute was not set in main.

Reply:
Idk, maybe `return self.inputfile` is enough, since in `mathics/core/definitions.py` it is already initialized to an empty string inside `__init__`

Reply:
That's true. So the initialization in the trivial case is not needed.